app/models/manageiq/providers/embedded_terraform/automation_manager/job.rb

```diff
@@ -26,24 +26,33 @@ def execute
     input_vars    = options.dig(:input_vars, :extra_vars) || {}
     action        = options[:action]
 
+    runner_options = {
+      :input_vars                  => decrypt_vars(input_vars),
+      :input_vars_type_constraints => input_vars_type_constraints,
+      :credentials                 => credentials,
+      :env_vars                    => options[:env_vars]
+    }
+
     case action
+    when ResourceAction::RECONFIGURE
+      Terraform::Runner.run(
+        Terraform::Runner::ActionType::UPDATE,
+        template_path,
+        runner_options.merge(:stack_id => options[:terraform_stack_id])
+      )
     when ResourceAction::RETIREMENT
-      Terraform::Runner.delete_stack(
-        options[:terraform_stack_id],
+      Terraform::Runner.run(
+        Terraform::Runner::ActionType::DELETE,
         template_path,
-        :input_vars                  => decrypt_vars(input_vars),
-        :input_vars_type_constraints => input_vars_type_constraints,
-        :credentials                 => credentials,
-        :env_vars                    => options[:env_vars]
+        runner_options.merge(:stack_id => options[:terraform_stack_id])
```

Comment:
Not for this PR, but if we could get the action dynamically from ResourceAction this could be a single case